backend/server.py

```diff
@@ -110,152 +178,182 @@ class Measurement(db.Model):
-            "name": hierarchy.name,
-            "description": hierarchy.description,
-        })
-    
-    all_hierarchies.reverse()
-
-    return jsonify(all_hierarchies), 200
+    return jsonify(201, new_node.to_dict())
 
 
-def get_measurements(hierarchy_id, node_id):
-    measurements = Measurement.query.filter_by(hierarchy_id=hierarchy_id, node_id=node_id)
-    measurement_list = []
+@app.route("/hierarchy/<hierarchy_id>/node/<node_id>/measurement", methods=['POST'])
+def create_measurement(hierarchy_id, node_id):
+    data = request.get_json()
 
-    for measurement in measurements:
-        new_measurement = {
-            "id": measurement.id,
-            "hierarchy_id": measurement.hierarchy_id,
-            "node_id": measurement.node_id,
+    new_measurement = Measurement.create(
+        data,
+        hierarchy_id=hierarchy_id,
+        node_id=node_id,
+    )
 
-            "name": measurement.name,
-            "type": measurement.type,
-            "value function": measurement.value_function,
-        }
-        measurement_list.append(new_measurement)
-    
-    return measurement_list
+    return jsonify(201, new_measurement.to_dict())
 
 
-def get_nodes(hierarchy_id, parent_id):
-    nodes = Node.query.filter_by(hierarchy_id=hierarchy_id, parent_id=parent_id)
-    node_list = []
+@app.route("/hierarchy/ascending_id", methods=['GET'])
+def get_all_hierarchies_ascending():
+    all_hierarchies = Hierarchy.get_list()
 
-    for node in nodes:
-        new_node = {
-            "id": node.id,
-            "hierarchy_id": node.hierarchy_id,
-            "parent_id": node.parent_id,
+    return jsonify(200, all_hierarchies)
 
-            "name": node.name,
-            "weight": node.weight,
-            "icon": node.icon,
-        }
-        
-        measurements_list = get_measurements(hierarchy_id, node.id)
-        nodes_list = get_nodes(hierarchy_id, node.id)
 
-        # Ensures the empty list is first on returned .json
-        if measurements_list == []:
-            new_node["measurements"] = measurements_list
-            new_node["children"] = nodes_list
-        else:
-            new_node["children"] = nodes_list
-            new_node["measurements"] = measurements_list
+@app.route("/hierarchy/descending_id", methods=['GET'])
+def get_all_hierarchies_descending():
+    all_hierarchies = Hierarchy.get_list()
+    all_hierarchies.reverse()
 
-        node_list.append(new_node)
-        
-    return node_list
+    return jsonify(200, all_hierarchies)
 
 
 @app.route("/hierarchy/<hierarchy_id>", methods=['GET'])
 def get_one_hierarchy(hierarchy_id):
-    hierarchy = Hierarchy.query.filter_by(id=hierarchy_id).first()
-    hier_dict = {
-        "id": hierarchy.id,
-        "name": hierarchy.name,
-        "description": hierarchy.description,
-    }
-
-    # Parse nodes and add list of dicts to hier_dict
-    hier_dict["nodes"] = get_nodes(hierarchy_id, None)
+    hier_dict = Hierarchy.get_dict(hierarchy_id)
```

Comment:
Should return a 404 if the hierarchy doesn't exist

Reply:
Added. However, I'm duplicating a lot of code. There may be a solution in changing the database model or something called a mixin.